source/game/kart/KartMove.cc

```diff
@@ -281,6 +292,91 @@ void KartMove::calc() {
     calcRotation();
 }
 
+/// @addr{0x80584334}
+void KartMove::calcRespawnStart() {
+    constexpr float RESPAWN_HEIGHT = 700.0f;
+
+    const auto *jugemPoint = System::RaceManager::Instance()->jugemPoint();
+    const EGG::Vector3f &jugemPos = jugemPoint->pos();
+    const EGG::Vector3f &jugemRot = jugemPoint->rot();
+
+    EGG::Vector3f respawnPos = jugemPos;
+    respawnPos.y += RESPAWN_HEIGHT;
+    EGG::Vector3f respawnRot = EGG::Vector3f(0.0f, jugemRot.y, 0.0f);
+
+    setInitialPhysicsValues(respawnPos, respawnRot);
+    state()->setTriggerRespawn(false);
+    state()->setInRespawn(true);
+}
+
+/// @addr{0x80579A50}
+void KartMove::calcInRespawn() {
+    constexpr f32 LAKITU_VELOCITY = 1.5f;
+    constexpr u16 RESPAWN_DURATION = 110;
+
+    if (!state()->isInRespawn()) {
+        return;
+    }
+
+    EGG::Vector3f newPos = pos();
+    newPos.y -= LAKITU_VELOCITY;
+    dynamics()->setPos(newPos);
+    dynamics()->setNoGravity(true);
+
+    if (++m_timeInRespawn > RESPAWN_DURATION) {
+        state()->setInRespawn(false);
+        state()->setAfterRespawn(true);
+        state()->setRespawnKillY(true);
+        m_timeInRespawn = 0;
+        m_flags.setBit(eFlags::Respawned);
+        dynamics()->setNoGravity(false);
+    }
+}
+
+/// @addr{0x80581C90}
+void KartMove::calcRespawnBoost() {
+    constexpr s16 RESPAWN_BOOST_DURATION = 30;
+    constexpr s16 RESPAWN_BOOST_INPUT_LENIENCY = 4;
+
+    if (!state()->isAfterRespawn()) {
+        if (m_respawnPostLandTimer <= 0) {
+            state()->setRespawnKillY(false);
+        } else if (state()->isAccelerateStart()) {
+            if (!state()->isBeforeRespawn()) {
+                activateBoost(KartBoost::Type::AllMt, RESPAWN_BOOST_DURATION);
+                m_respawnTimer = RESPAWN_BOOST_DURATION;
+            }
+
+            m_respawnPostLandTimer = 0;
+        }
+
+        m_respawnPostLandTimer = std::max(0, m_respawnPostLandTimer - 1);
```

Comment:
Pedantically, the `m_respawnPostLandTimer` management must go in the broader `else` statement, but is not specific to if we are or aren't pressing in the accelerate start state. In practice, the state's the same and the code is one less layer nested. Dealer's choice on if you want to ignore this review comment.

Reply:
Hmm I think I'm a bit confused. See my updated changes. `m_respawnPostLandTimer` occurs regardless of whether or not  we are in accelerate start state, but `m_respawnPostLandTimer  = 0` should only occur if we are in accelerate start state.